actions/river/delete.php

```diff
@@ -6,14 +6,23 @@
  * @subpackage Core
  */
 
-$id = get_input('id', false);
+$id = (int)get_input('id');
 
-if ($id !== false && elgg_is_admin_logged_in()) {
-	if (elgg_delete_river(array('id' => $id))) {
-		system_message(elgg_echo('river:delete:success'));
-	} else {
-		register_error(elgg_echo('river:delete:fail'));
-	}
+$items = elgg_get_river(['id' => $id]);
+if (!$items) {
+	register_error(elgg_echo('river:delete:fail'));
+	forward(REFERER);
+}
+
+$item = $items[0];
+
+if (!$item->canDelete()) {
+	register_error(elgg_echo('river:delete:lack_permission'));
+	forward(REFERER);
+}
+
+if (elgg_delete_river(array('id' => $item->id))) {
```

Comment:
I guess I doubt people are using elgg_delete_river to removing tons of items anyway. Better to have a consistent API.